test/test_classifier.py

```diff
@@ -46,67 +46,112 @@ def test_classifier_drops_features():
     for column in dropped_columns:
-        assert {column}.issubset(frames_db_new) == False
-    assert {'data_rate'}.issubset(frames_db_new.columns) == True
+        assert {column}.issubset(new_frames_db) == False
+
+
+def test_classifier_correct_null_values():
+    cl = Classifier(1)
+    frames_db = frame_factory(1).to_dataframe()
+    for i in range(1, 20):
+        frames_db = pd.concat([frames_db, frame_factory(i).to_dataframe()], axis=0)
+    new_frames_db = cl.drop_features(frames_db)
+
+    assert {'data_rate'}.issubset(new_frames_db.columns) == True
     assert frames_db['data_rate'].isnull().values.any() == True
-    assert frames_db_new['data_rate'].isnull().values.any() == False
+    for i, e in enumerate(frames_db['data_rate'].values.tolist()):
+        if e == None:
+            assert new_frames_db['data_rate'].values.tolist()[i] == 0
 
 
 @pytest.fixture
 def cl():
-    cl = Classifier(1)
+    cl = Classifier(2)
 
-    frames_db = frame_factory(1).to_dataframe()
+    frames_df = frame_factory(1).to_dataframe()
     for i in range(1, 20):
-        frames_db = pd.concat([frames_db, frame_factory(i).to_dataframe()], axis=0)
+        frames_df = pd.concat([frames_df, frame_factory(i).to_dataframe()], axis=0)
 
     labels = pd.DataFrame.from_dict({'Address': ["00:00:00:00:00:01"], 'Label': ['test']})
-    training_data, label_series = cl.preprocess_data(frames_db, labels)
+    training_data, label_series = cl.preprocess_data(frames_df, labels)
 
     cl.model.fit(training_data, label_series)
     return cl
 
 
-def test_classifier_extract_features(cl):
-    assert True
+def test_classifier_has_labels(cl):
+    assert cl.labels_in_model()[0] == 'test'
 
 
-def test_classifier_has_labels(cl):
-    assert cl.labels_in_model()[0] is 'test'
+def generator(items: list):
+    for item in items:
+        yield item
 
 
-def test_classifier_returns_label_for_interval(cl):
-    frames = []
-    for i in range(0, 10):
-        frames.append(frame_factory(i))
-    possible_labels_list = cl.labels_in_model()
+def test_classifier_accumulate_frames():
+    cl = Classifier(2)
+    frame_gen = generator([frame_factory(0), frame_factory(0.5), frame_factory(1), frame_factory(1.9),
+                           frame_factory(2), frame_factory(2.1), frame_factory(2.5), frame_factory(3)])
+    accumulated_frames = next(cl.accumulate_frames(frame_gen))
+    for e in accumulated_frames:
+        assert isinstance(e, WifiFrame)
+    assert len(accumulated_frames) == 4
 
-    assert cl.classify_interval_label(frames) in possible_labels_list
+
+def test_classifier_returns_label_for_classify(cl):
+    frame_gen = generator([frame_factory(1), frame_factory(2), frame_factory(3), frame_factory(4)])
+    possible_labels_list = cl.labels_in_model()
+    assert next(cl.classify(frame_gen)) in possible_labels_list
 
 
-def test_classifier_return_confidence_for_interval(cl):
+@pytest.fixture
+def frames():
     frames = []
```

Comment:
Ikke noget du skal ændre, men vil bare lige vise hvad python også kan :)
```suggestion
    frames = [frame_factory(i) for i in range(1,20)]
```

Reply:
Noteret, takker ;)